src/ucp/stream/stream_recv.c

```diff
@@ -232,16 +232,13 @@ static UCS_F_ALWAYS_INLINE void
 ucp_stream_recv_request_init(ucp_request_t *req, ucp_ep_h ep, void *buffer,
                              size_t count, size_t length,
                              ucp_datatype_t datatype,
-                             ucp_stream_recv_callback_t cb,
+                             const ucp_request_param_t *param,
                              uint32_t request_flags)
 {
-    req->flags              = UCP_REQUEST_FLAG_CALLBACK |
-                              UCP_REQUEST_FLAG_STREAM_RECV |
-                              request_flags;
+    req->flags              = UCP_REQUEST_FLAG_STREAM_RECV | request_flags;
```

Comment:
i'd remove `request_flags` parameters, looks like all needed info is in params now